src/ui/public/courier/search_source/search_source.js

```diff
@@ -105,14 +105,14 @@ const GETTABLE_PROPS = [
   'index', // This is settable via `setIndexPattern`
 ];
 
-function parseInitialState(initialState) {
-  if (!initialState) {
+function parseInitialData(initialData) {
```

Comment:
I personally think that "data" is too ambiguous. While `getData()` is better than `toJSON()`, I think `getState()`, `getAllValues()`, or something similar is more of an improvement. "data" can really just be anything, at least "state" is clearly linked to the state of the get/set's that this object supports. "values" mirrors the naming of the `getValue()` and `setValue()` methods. But the "data" could literally be anything IMO.